src/nb_enum.cpp

```diff
@@ -41,15 +41,25 @@ static PyObject *nb_enum_lookup(PyObject *self) {
 }
 
 static PyObject *nb_enum_repr(PyObject *self) {
-    PyObject *entry = nb_enum_lookup(self);
-    if (!entry)
-        return nullptr;
-
     PyObject *name = nb_inst_name(self);
-    PyObject *result =
-        PyUnicode_FromFormat("%U.%U", name, NB_TUPLE_GET_ITEM(entry, 0));
+    PyObject *entry = nb_enum_lookup(self);
+    PyObject *result;
+    if (!entry) {
+        PyErr_Clear();
+        enum_supplement &supp = nb_enum_supplement(Py_TYPE(self));
+        PyObject *int_val = supp.is_signed ? nb_enum_int_signed(self)
+                                           : nb_enum_int_unsigned(self);
+        if (!int_val) {
+            Py_DECREF(name);
+            return nullptr;
+        }
+        result = PyUnicode_FromFormat("%U(%S)", name, int_val);
```

Comment:
This is not conditional on `allow_unenumerated`, because you can still get unenumerated values without that annotation by returning them from C++, and it's a pretty bad debugging experience when `repr()` throws.